generators/triangle.go

```diff
@@ -0,0 +1,45 @@
+package generators
+
+import (
+	"errors"
+	"math"
+
+	"github.com/faiface/beep"
+)
+
+type triangleGenerator struct {
+	dt float64
+	t  float64
+}
+
+// Creates a streamer which will procude an infinite triangle wave with the given frequency.
+// use other wrappers of this package to change amplitude or add time limit.
+// sampleRate must be at least two times grater then frequency, otherwise this function will return an error.
+func TriangleTone(sr beep.SampleRate, freq float64) (beep.Streamer, error) {
+	dt := freq / float64(sr)
+
+	if dt >= 1.0/2.0 {
+		return nil, errors.New("faiface triangle tone generator: samplerate must be at least 2 times grater then frequency")
+	}
+
+	return &triangleGenerator{dt, 0}, nil
+}
+
+func (g *triangleGenerator) Stream(samples [][2]float64) (n int, ok bool) {
+	for i := range samples {
+		if g.t < 0.5 {
+			samples[i][0] = 2.0 * g.t
+			samples[i][1] = 2.0 * g.t
+		} else {
+			samples[i][0] = 2.0 * (1.0 - g.t)
+			samples[i][1] = 2.0 * (1.0 - g.t)
```

Comment:
```suggestion
			samples[i][0] = 2.0*(1-g.t) - 1
			samples[i][1] = 2.0*(1-g.t) - 1
		} else {
			samples[i][0] = 2.0*g.t - 1.0
			samples[i][1] = 2.0*g.t - 1.0
```

First line = before
Second line = after

<img width="998" alt="image" src="https://user-images.githubusercontent.com/77609814/156284115-d7292e4d-44b3-4f9a-9c70-48b0b7a0f3a4.png">

Reply:
I'm not sure if we want the triangle to starts at the 0 value 🤔 
This could prevent a "clicky sound" when switching from one osc to another... (but I think for this example it is a detail) ;-)

<img width="49" alt="image" src="https://user-images.githubusercontent.com/77609814/156284324-e9664f6c-a826-4404-beb7-f71b68d88f0f.png">